src/Microsoft.Restier.Core/Query/DefaultQueryHandler.cs

```diff
@@ -58,7 +58,7 @@ internal static class DefaultQueryHandler
 
             // execute query
             QueryResult result;
-            var executor = context.GetApiService<IQueryExecutor>();
+            var executor = context.GetApiContextService<IQueryExecutor>();
```

Comment:
Again I'm inclined to retrieve services from the InvocationContext scope, scopes are equally created, ApiContext scope isn't special, when compared to InvocationContext scope.